File: ts/src/36.ts
//3417. Zigzag Grid Traversal With Skip

function zigzagTraversal(grid: number[][]): number[] {
  let result: number[] = [];
  let n = grid.length; // Number of rows
  if (n === 0) return result;
  let m = grid[0].length; // Number of columns

  for (let i = 0; i < n; ++i) {
    if (i % 2 === 0) {
      // Left to right
      for (let j = 0; j < m; ++j) {
        if (j % 2 === 0) {
          // Include only even-indexed cells
          result.push(grid[i][j]);
        }
      }
    } else {
      // Right to left
      for (let j = m - 1; j >= 0; --j) {
        if (j % 2 !== 0) {
          // Include only odd-indexed cells
          result.push(grid[i][j]);
        }
      }
    }
  }

  return result;
}

// sol: https://leetcode.com/problems/zigzag-grid-traversal-with-skip/solutions/6267543/3417-zigzag-grid-traversal-with-skip40-b-9vq0/
// sub: https://leetcode.com/problems/zigzag-grid-traversal-with-skip/submissions/1505790005/

File: rust/src/36.rs
//36. Valid Sudoku
impl Solution{
    pub fn is_valid_sudoku(board: Vec<Vec<char>>) -> bool {
        let mut rowcase = [[0; 9]; 9];
        let mut colcase = [[0; 9]; 9];
        let mut gridcase = [[0; 9]; 9];

        for i in 0..9 {
            for j in 0..9 {
                let c = board[i][j];
                if c != '.' {
                    let num = (c as u8 - b'1') as usize; // convert '1'-'9' to 0-8
                    let grid = (i / 3) * 3 + (j / 3);

                    if rowcase[i][num] > 0 || colcase[j][num] > 0 || gridcase[grid][num] > 0 {
                        return false;
                    }

                    rowcase[i][num] += 1;
                    colcase[j][num] += 1;
                    gridcase[grid][num] += 1;
                }
            }
        }

        true
    }
}
//sub: https://leetcode.com/problems/valid-sudoku/submissions/1753048692/
//sol: https://leetcode.com/problems/valid-sudoku/solutions/5570731/36-valid-sudoku/?envType=daily-question&envId=2025-08-30
